pkg/vcs/gitlab/client.go

```diff
@@ -129,15 +129,25 @@ func (c *GitlabClient) GetOldRunUrls(mrIID int, project string, rootNoteID int)
 	for _, note := range notes {
 		if note.Author.Username == currentUser.Username {
 			runUrl := utils.CaptureSubstring(note.Body, utils.URL_RUN_PREFIX, utils.URL_RUN_SUFFIX)
+			// We scrape the run URLs from the previous MR comments.
+			// Since they are hyperlinked in markdown format, we need to extract the URL
+			// without the markdown artifacts.
+			runUrlRaw := utils.CaptureSubstring(runUrl, "[", "]")
+			runUrlSplit := strings.Split(runUrlRaw, "/")
+			// The run ID is the last part of the run URL, and it looks like run-abcd12345...
+			runID := runUrlSplit[len(runUrlSplit)-1]
```

Comment:
We should have a check to ensure that runUrlSplit > 0 so it won't panic

Reply:
Thanks for catching that - I also added a else condition in the event the runUrlSplit check fails to just fallback to original URL that was used before just so the has _something_ rather than just an awkward blank table entry